Experimental/DistributedAuthoritySample/Assets/Scripts/UI/IngameUI/PlayersTopUIController.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using System.Collections.Generic;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace Unity.Multiplayer.Samples.SocialHub.UI
+{
+    /// <summary>
+    /// UI controller which displays nameplate and mic icon on top of each connected player.
+    /// </summary>
+    public class PlayersTopUIController : MonoBehaviour
+    {
+        [SerializeField]
+        UIDocument m_UIDocument;
+
+        [SerializeField]
+        VisualTreeAsset m_NameplateAsset;
+
+        [SerializeField]
+        float m_PanelMinSize = 0.8f;
+
+        [SerializeField]
+        float m_PanelMaxSize = 1.1f;
+
+        [SerializeField]
+        float m_DisplayYOffset = 1.3f;
+
+        [SerializeField]
+        Camera m_Camera;
+
+        List<PlayerHeadDisplay> m_PlayerHeadDisplayPool = new();
+
+        List<Tuple<NetworkObject, PlayerHeadDisplay>> m_PlayersToDisplayMap = new();
+
+        VisualElement m_Root;
+
+        const int k_PoolSize = 12;
+
+        void OnEnable()
+        {
+            m_PlayerHeadDisplayPool = new List<PlayerHeadDisplay>();
+            for (var i = 0; i < k_PoolSize; i++)
+            {
+                m_PlayerHeadDisplayPool.Add(new PlayerHeadDisplay(m_NameplateAsset));
+            }
+
+            m_Root = m_UIDocument.rootVisualElement.Q<VisualElement>("player-top-display-container");
+        }
+
+        private void Update()
```

Comment:
I'd refactor this code to only include the part where the VisualElements are moved in space, and not the adding/removing of players, since that's going to be invoked from client-side scripts on connection and disconnection.
So, so long as this class has methods to add and remove player displays, that's fine for now.
It's fine if nothing is displayed at the moment since we'll do an integration PR will follow this up.